plugins/extractors/bigquery/bigquery.go

```diff
@@ -164,22 +167,31 @@ func (e *Extractor) Init(ctx context.Context, config plugins.Config) error {
 
 // Extract checks if the table is valid and extracts the table schema
 func (e *Extractor) Extract(ctx context.Context, emit plugins.Emit) error {
+	pageSize := e.config.MaxPageSize
+	if e.config.DatasetPageSize != 0 {
+		pageSize = e.config.DatasetPageSize
+	}
+
 	// Fetch and iterate over datasets
-	it := e.client.Datasets(ctx)
-	it.PageInfo().MaxSize = e.getMaxPageSize()
+	pager := iterator.NewPager(e.client.Datasets(ctx), pageSize, "")
 	for {
-		ds, err := it.Next()
-		if errors.Is(err, iterator.Done) {
-			break
-		}
+		var datasets []*bigquery.Dataset
+		nextToken, err := pager.NextPage(&datasets)
 		if err != nil {
 			return fmt.Errorf("fetch dataset: %w", err)
 		}
-		if IsExcludedDataset(ds.DatasetID, e.config.Exclude.Datasets) {
-			e.logger.Debug("excluding dataset from bigquery extract", "dataset_id", ds.DatasetID)
-			continue
+
+		if errors.Is(err, iterator.Done) || nextToken == "" {
```

Comment:
Don't think we need to check `iterator.Done` here. Also, we should have the `nextToken` check at the end of the loop. There might not be a next page but we could have gotten some results in the current page.

Reply:
similar change needed for tables pager as well.